### ISSUE TITLE
[BUG] Backend User gets logged out when editing page content on a hidden page beneath a Sys-Folder

### ISSUE THREAD

**Issue by phth** [TYPO3-Solr/ext-solr#3702]
Given the following page tree:

![image](https://github.com/TYPO3-Solr/ext-solr/assets/1274061/2cb966bf-3d77-4e93-8545-cab0a80c67d2)

the backend user (admin or/and editor) that is trying to create or edit a content element on the page is logged out after saving the changes.
The behaviour is the same as described in #3246 

I was able to track it down in a similar way:

1. `ApacheSolrForTypo3\Solr\FrontendEnvironment\Tsfe->initializeTsfe()`
2. `$tsfe->determineId($serverRequest)`
3. `GeneralUtility::makeInstance(ErrorController::class)->pageNotFoundAction()`

After looking into pageNotFoundAction() I tracked it down to `\TYPO3\CMS\Core\Error\PageErrorHandler\PageContentErrorHandler->handlePageError()`

There, `\TYPO3\CMS\Core\Error\PageErrorHandler\PageContentErrorHandler->sendSubRequest()` is called, wich causes another reinitialisation of TSFE that may be messing up other $GLOBALS variables that are not backed up before like $GLOBALS['BE_USER'].

Only if the subrequest is created, the current backend user gets logged out, so instead of tracking down the root cause even further (to second initialization of tsfe / running through all middlewares) I implemented a workaround:

I created a custom PageContentErrorHandler that tries to check if the created TSFE is a solr-initialized one: https://gist.github.com/phth/fdb2ac5f648e5b73b130306320ba7f10
and configured it in config.yaml:

```
errorHandling:
  -
    errorCode: '404'
    errorHandler: PHP
    errorPhpClassFQCN: Webicorns\SitePackage\Error\PageContentErrorHandler
    errorContentSource: 't3://page?uid=637'
```

From this, I have made the following conclusions:

- solr is not really responsible for the behaviour
- the behaviour is implemented in TYPO3 core due to the internal subrequests
- the former workaround, described in #3246 - setting the feature flag subrequestPageErrors to false - is not an option anymore because it was removed in TYPO3 12.0, see https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Breaking-96107-DeprecatedFunctionalityRemoved.html
- the error may not occure if someone already uses a custom ErrorHandler

**Used versions**
 - TYPO3 Version: 12.4.2
 - EXT:solr Version: dev-main / f4d1395

**Comment by dmitryd** [comment]
I cannot reproduce it in the v12 branch.

**Comment by dmitryd** [comment]
And I cannot reproduce it with solr 11.5.0 / TYPO3 v11 either. What I did in both cases:

- Hide a page in the page tree using the context menu
- Go to that page
- Create a new content element & save it
- Edit the content element & save it

I am still logged in.

**Comment by PLACEBOBRO** [comment]
This bug is also reproduceable by me. I'm using TYPO3 12.4.2 and EXT:solr Version dev-main / 9886e74 and if I add a new page under website root, the backend user gets logged out.

**Comment by dmitryd** [comment]
I am working in v12 branch. My `composer.json` is like:

```json
		"apache-solr-for-typo3/solr": "dev-task/3376-TYPO3_12_compatibility"
```

It is not yet merged to `main`. Are you all sure that you need to use `main`?

**Comment by dkd-kaehm** [comment]
> I am working in v12 branch. My `composer.json` is like:
> 
> ```json
> 		"apache-solr-for-typo3/solr": "dev-task/3376-TYPO3_12_compatibility"
> ```
> 
> It is not yet merged to `main`. Are you all sure that you need to use `main`?

Yes, the main branch is right branch. 
task/3376-TYPO3_12_compatibility was there, becouse alphas were tagged from this branch.
I'll remove the task/3376-TYPO3_12_compatibility soonish.

**Comment by dmitryd** [comment]
Okay. Then the bug is new because it does not happen in that branch :(

**Comment by b13-michaelsemle** [comment]
I got the same ish issue like phth and PLACEBOBRO .
Reproduce:
- TYPO3 v12.4.1
- Using EXT version 12.0.0-beta-1
- Creating a page via drag&drop underneath a sys_folder with hidden = 0
- Strage that with sys_folder and hidden = 1 everything is fine

After debug: seems to be around the record monitoring. Changing record monitoring to 2 fixes the logout bug.

**Comment by dmitryd** [comment]
@b13-michaelsemle 

> Creating a page via drag&drop underneath a sys_folder with hidden = 0

It creates a page, which is hidden. As long as the page gets a hidden flag, the problem happens. You can simply set any page inside the folder to hidden via context menu and you are logged out at once.

**Comment by mkarulin** [comment]
I can reproduce this error on SOLR dev-main branch and TYPO3 12.4.3

**Comment by dkd-kaehm** [comment]
Fixed by #3720, thanks @dmitryd.
I'll close that issue, because it is not reproducible from description and comments on Version: https://github.com/TYPO3-Solr/ext-solr/commit/bbdda0cf2dfc946d09b5207c58ece034696cf63e.

Please open a new issue with exact description how to reproduce within https://github.com/TYPO3-Solr/solr-ddev-site
or providing a pull-request with a test-case.
